compile.ts

```diff
@@ -341,15 +341,15 @@ class Compiler {
           }
         }
 
-        this.#emitLabel(head);
-        if (ts.isWhileStatement(n)) {
-          this.#emitLabel(cont);
+        if (ts.isDoStatement(n)) {
+          this.#jump(body);
         }
+        this.#emitLabel(head);
         let cond = ts.isForStatement(n)
           ? n.condition
-          : ts.isWhileStatement(n)
+          : ts.isWhileStatement(n) || ts.isDoStatement(n)
           ? n.expression
-          : undefined;
+          : undefined
```

Comment:
Couldn't this just be `let cond = ts.isForStatement(n) ? n.condition : n.expression` now?

Reply:
Done. Was wrongly thinking the type system would be confused.